server/sonar-server/src/main/java/org/sonar/server/computation/task/projectanalysis/filemove/FileMoveDetectionStep.java

```diff
@@ -238,8 +268,19 @@ private File getFile(DbSession dbSession, DbComponent dbComponent) {
 
   private static void printIfDebug(ScoreMatrix scoreMatrix) {
     if (LOG.isDebugEnabled()) {
-      LOG.debug("ScoreMatrix:\n" + scoreMatrix.toCsv(';'));
+      try {
+        Path tempFile = Files.createTempFile("score-matrix", ".csv");
+        try (BufferedWriter writer = Files.newBufferedWriter(tempFile, Charset.forName("UTF-8"))) {
+          writer.write(scoreMatrix.toCsv(';'));
+        }
+        LOG.info("score matrix dumped as CSV: {}", tempFile);
+      } catch (IOException e) {
+        e.printStackTrace();
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Use a logger to log this exception. ([squid:S1148](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1148))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fsl%2FSONAR-10253%2Foptimize_file_move_detection_based_on_file_size&issues=AWGVbbwdRRutcW4uJrJG&open=AWGVbbwdRRutcW4uJrJG)

[ISSUE_KEY]: AWGVbbwdRRutcW4uJrJG